WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/BlogListReducer.swift

```diff
@@ -0,0 +1,184 @@
+enum BlogListReducer {
```

Comment:
I do not see why this isn't a struct or a class. Instead of injecting a `UserPersistentRepository` in each function, we can turn this into a struct and inject it in the initializer. Having this as an enum hurts the readability as well. What do you think?

Reply:
The goal was to keep the functions static so that we can have stateless functions. I don't mind refactoring. Likely due to your request above, I'll need to hold state anyway.

Reply:
Resolved in https://github.com/wordpress-mobile/WordPress-iOS/pull/22833/commits/0c91c5df027ca82901f837cdb08f31b8792570ee